war/src/main/java/io/lumeer/engine/rest/CollectionService.java

```diff
@@ -217,6 +217,38 @@ public void dropCollection(final @PathParam("collectionCode") String collectionC
    }
 
    /**
+    * Returns a collection from the database by it's code.
+    * @param collectionCode
+    *       collection code
+    * @throws CollectionNotFoundException
+    *       When the given collection does not exist.
+    * @throws UnauthorizedAccessException
+    *       When current user is not allowed to read the collection.
+    * @return Collection with provided code
+    */
+   @GET
+   @Path("/{collectionCode}/")
+   public Collection getCollection(final @PathParam("collectionCode") String collectionCode) throws CollectionNotFoundException, UnauthorizedAccessException {
+      if (collectionCode == null) {
+         throw new BadRequestException();
+      }
+
+      if (!collectionFacade.hasCollection(collectionCode)) {
+         throw new UserCollectionNotFoundException(ErrorMessageBuilder.userCollectionNotFoundString(collectionCode));
+      }
+
+      if (!securityFacade.hasCollectionRole(projectCode, collectionCode, LumeerConst.Security.ROLE_READ)) {
+         throw new UnauthorizedAccessException();
+      }
+
+      return collectionFacade.getCollections().stream()
+                             .filter(collection -> securityFacade.hasCollectionRole(projectCode, collection.getCode(), LumeerConst.Security.ROLE_READ))
+                             .filter(collection -> collection.getCode().equals(collectionCode))
+                             .findFirst()
+                             .orElseThrow(() -> new CollectionNotFoundException(ErrorMessageBuilder.userCollectionNotFoundString(collectionCode)));
+   }
```

Comment:
You should add new method to `CollectionFacade` to read collection directly, because reading all collections is very unefficient.